test-tools/src/main/java/org/terracotta/utilities/test/logging/ConsoleAppenderCapture.java

```diff
@@ -0,0 +1,235 @@
+   */
+  public ConsoleAppenderCapture(String testId) {
+    this.testId = (testId == null ? "present" : testId);
+
+    MDC.put(mdcKey, this.testId);
+    Runnable removeAppenders = () -> {
+    };
+    LoggerContext context = (LoggerContext)LoggerFactory.getILoggerFactory();
+
+    /*
+     * Scan for all loggers using the ConsoleAppender and add an appender to capture
+     * the output for the designated ConsoleAppender target.
+     */
+    for (ch.qos.logback.classic.Logger logger : context.getLoggerList()) {
+      Iterator<Appender<ILoggingEvent>> appenderIterator = logger.iteratorForAppenders();
+      while (appenderIterator.hasNext()) {
+        Appender<ILoggingEvent> iLoggingEventAppender = appenderIterator.next();
+        if (iLoggingEventAppender instanceof ConsoleAppender) {
+          String target = ((ConsoleAppender<?>)iLoggingEventAppender).getTarget();
+          ListAppender<ILoggingEvent> appender = appenderMap.computeIfAbsent(target, t -> {
+            ListAppender<ILoggingEvent> listAppender = new ListAppender<>();
+            listAppender.setContext(context);
+
+            listAppender.addFilter(new AbstractMatcherFilter<ILoggingEvent>() {
+              @Override
+              public FilterReply decide(ILoggingEvent iLoggingEvent) {
+                if (iLoggingEvent.getMDCPropertyMap().getOrDefault(mdcKey, "").equals(ConsoleAppenderCapture.this.testId)) {
+                  return FilterReply.ACCEPT;
+                } else {
+                  return FilterReply.DENY;      // Filter only applies to 'this' Appender -- can stop processing here
+                }
+              }
+            });
+
+            listAppender.start();
+            return listAppender;
+          });
+          logger.addAppender(appender);
+          Runnable r = removeAppenders;
+          removeAppenders = () -> {
+            r.run();
+            logger.detachAppender(appender);
+            appender.stop();
+          };
+          break;
+        }
+      }
+    }
+
+    this.removeAppenders = removeAppenders;
+  }
+
+  /**
+   * Gets a reference to the {@code List}s into which log entries are captured.
+   *
+   * @return the map of logging target to log entry list reference; the key of the
+   *      map is the {@link ConsoleAppender#getTarget()} value for which
+   *      {@link Target#targetName()} value may be used
+   * @see Target
+   */
+  public Map<String, List<ILoggingEvent>> getLogs() {
+    return appenderMap.entrySet().stream()
+        .collect(toMap(Map.Entry::getKey, e -> e.getValue().list));
+  }
+
+  /**
+   * Gets the events, in formatted string form, logged to the specified target.
+   * @param target the target for which log messages are to be returned
+   * @return the list of messages logged to {@code target}; may be empty
+   */
+  public List<String> getMessages(Target target) {
+    requireNonNull(target, "target");
+    return getLogs().getOrDefault(target.targetName(), Collections.emptyList())
```

Comment:
Seems a bit wasteful to stream and collect the whole map... and then throw away everything but the one key? Might be nicer to just access `appenderMap` directly?